edx/edx-app-ios
# Shift Due Dates from Assignment View

### Description

[LEARNER-7857](https://openedx.atlassian.net/browse/LEARNER-7857)

As a mobile learner in a self-paced course, I want to shift my schedule dates from my assignments so I can submit assignments that were previously past due

- Display banner message in assignments that are past due with “shift due dates” button 

- Display “shift due dates” link beside assignments' disabled submit buttons with info tooltip

- Display success toast message when learner shifts dates from either the banner button or link, with the following text:

“Your due dates have been successfully shifted to help you stay on track.”

- Display link to Dates page in success message with the text “View all dates” _(currently hidden as deeplinking is not available)_

- Clicking this link brings user to Dates page _(currently hidden as deeplinking is not available)_

- Success toast message should also appear when learner shifts date from Course home or Dates page

When learner shifts their due dates, previously disabled assignment “Submit” buttons should become enabled

### How to test this PR

- return `.resetDatesBanner` from following lines, banner will appear

- Click on shift button

- Screen will be reloaded

- Submit button will be enabled inside webview

https://github.com/edx/edx-app-ios/blob/7badb1de291fab74d592bd62e65e4dd379f4c522/Source/CourseDates.swift#L212-L224

### Screenshot
<img width="370" alt="Simulator Screen Shot - iPhone 11 Pro Max - 2020-10-26 at 11 02 30" src="https://user-images.githubusercontent.com/960241/97138784-cb592c80-177a-11eb-804e-8f117d979c5a.png">
